internal/controller/dnsrecord_controller.go

```diff
@@ -81,9 +81,14 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	dnsRecord := previous.DeepCopy()
 
 	if dnsRecord.DeletionTimestamp != nil && !dnsRecord.DeletionTimestamp.IsZero() {
-		if err = r.deleteRecord(ctx, dnsRecord); err != nil {
+		requeueTime, err := r.deleteRecord(ctx, dnsRecord)
+		if err != nil {
 			logger.Error(err, "Failed to delete DNSRecord")
-			return ctrl.Result{}, err
+			return ctrl.Result{RequeueAfter: requeueTime}, err
+		}
+		// requeue for validation
+		if requeueTime == validationRequeueTime {
```

Comment:
Ok so delete was successful but we need to revalidate. Hmm this took me a while to figure out. Could we add a comment explaining that is what is happening here and why we do this comparison?